src/ray/common/task/task_spec.h

```diff
@@ -180,10 +181,13 @@ class TaskSpecification : public MessageWrapper<rpc::TaskSpec> {
   /// Cached scheduling class of this task.
   SchedulingClass sched_cls_id_;
 
+  static absl::Mutex mutex_;
```

Comment:
Please add a comment explaining why this is needed and what race conditions it protects against.

Reply:
Done.